### PR TITLE
[MEN-25] - Dispatch move requests after routing config change.

### PR DESCRIPTION
**Pull request by dalloriam**
The main function (the one which does the stateless identification of pending moves) needs a bit of love to refactor it more, but it'll have to wait until _after_ the big upcoming node refactor.

### CODE REVIEW

File: bin/menmosd/tests/menmosd_routing.rs [hunk 1]
```diff
@@ -1,7 +1,10 @@
 //! Test blob routing.
+use std::net::IpAddr;
```
Comment (laurentlp) [inline bin/menmosd/tests/menmosd_routing.rs:2]: Remove this.

File: bin/menmosd/src/menmosd/node/node_impl.rs [hunk 2]
```diff
@@ -158,19 +166,100 @@ where
...
+        let mut move_requests = Vec::with_capacity(MOVE_REQUEST_BATCH_SIZE);
+
+        for username in self.index.users().iter().filter_map(|f| f.ok()) {
+            let routing_config_maybe = self.index.routing().get_routing_config(&username)?;
+            if routing_config_maybe.is_none() {
+                continue;
+            }
+
+            let mut routing_config_state = routing_config_maybe.unwrap();
+            if routing_config_state.state == DirtyState::Clean {
```
Comment (gabrielmethot) [inline bin/menmosd/src/menmosd/node/node_impl.rs:204]: `DirtyState::Clean` sounds confusing without deeper knowledge of the code

File: bin/menmosd/src/menmosd/node/test/mock.rs [hunk 3]
```diff
@@ -137,6 +141,10 @@ impl RoutingMapper for MockRoutingMap {
         guard.remove(username);
         Ok(())
     }
+
+    fn iter(&self) -> DynIter<'static, Result<RoutingConfigState>> {
+        unimplemented!()
```
Comment (gabrielmethot) [inline bin/menmosd/src/menmosd/node/test/mock.rs:146]: Shame on you... Add a TODO?

File: bin/menmosd/src/menmosd/node/test/mock.rs [hunk 4]
```diff
@@ -383,6 +391,11 @@ impl UserMapper for MockUserMap {
         let guard = self.users.lock().unwrap();
         Ok(guard.contains_key(username))
     }
+
+    fn iter(&self) -> DynIter<'static, Result<String>> {
+        // Returning an iterator on something protected by a mutex = cursed.
+        unimplemented!();
```
Comment (gabrielmethot) [inline bin/menmosd/src/menmosd/node/test/mock.rs:397]: Same as above.

File: bin/menmosd/tests/menmosd_routing.rs [hunk 5]
```diff
@@ -34,3 +37,47 @@ async fn get_set_delete_routing_config() -> Result<()> {
...
+    let blob_a = cluster
+        .push_document(
+            "yeet yeet",
+            Meta::file("file1.txt").with_meta("some_field", "bing"),
+        )
+        .await?;
+
+    let blob_b = cluster
+        .push_document(
+            "bing bong",
```
Comment (gabrielmethot) [inline bin/menmosd/tests/menmosd_routing.rs:56]: Should have been "bwing bwoom" for sure

**Review by laurentlp — APPROVED** [review]

**Pull request merged**